docs/userguide/searchspace.md

```diff
@@ -1,46 +1,247 @@
+subspace = continuous = SubspaceContinuous.from_bounds(bounds)
+```
+
+### Constructing from a dataframe
+
+Similar to discrete subspaces, continuous spaces can also be constructed using [`SubspaceContinuous.from_dataframe`](baybe.searchspace.continuous.SubspaceContinuous.from_dataframe).
+However, when using this method to create a continuous space, it will create the smallest axis-aligned hyperrectangle-shaped continuous subspace that contains the points specified in the given dataframe.
+
+```python
+from baybe.parameters import NumericalContinuousParameter
+from baybe.searchspace.continuous import SubspaceContinuous
+
+points = pd.DataFrame(
+    {
+        "param1": [0, 1, 2],
+        "param2": [-1, 0, 1],
+    }
+)
+subspace = SubspaceContinuous.from_dataframe(df=points)
+```
+
+As for discrete subspaces, this method automatically infers the parameter types.
+
+~~~
+ Continuous Parameters
+      Name                          Type  Lower_Bound  Upper_Bound
+ 0  param1  NumericalContinuousParameter          0.0          2.0
+ 1  param2  NumericalContinuousParameter         -1.0          1.0
+~~~
+
+## Constructing full search spaces
+
+There are several methods available for creating full search spaces.
 
-A search space is constructed by providing either a discrete subspace, a continuous subspace, or both of them. 
+### From the default constructor 
 
-## Understanding the different search spaces and subspaces
+It is possible to construct a discrete search space by simply using the default constructor of the `SearchSpace` class.
+The required parameters are derived from the `__init__` function of that class.
+In the simplest setting, it is sufficient to provide a `SubspaceDiscrete` in order to create a discrete subspace.
 
-In general, subspaces are managed via the respective classes, that is, via [`SubspaceDiscrete`](baybe.searchspace.discrete.SubspaceDiscrete) and [`SubspaceContinuous`](baybe.searchspace.continuous.SubspaceContinuous). They are built from parameter definitions and optional constraints, and provide access to candidate sets and different parameter views.
+```python
+from baybe.parameters import NumericalDiscreteParameter
+from baybe.searchspace import SearchSpace, SubspaceDiscrete
 
-Both kinds of subspaces share the following attributes:
-* **A parameter list:** The list of the parameters of the subspace. Note that they need to be discrete resp. continuous, depending on the subspace (see [here](./parameters) for details.)
-* **Optional constraint list(s):** A single resp. up to two lists can be provided for general constraints in discrete subspaces resp. linear equality and linear in-equality constraints in continuous subspaces. For more details on constraints, see [here](./constraints).
+parameters = [
+    NumericalDiscreteParameter(name="x0", values=[1, 2, 3]),
+    NumericalDiscreteParameter(name="x1", values=[4, 5, 6]),
+    NumericalDiscreteParameter(name="x2", values=[7, 8, 9]),
+]
```

Comment:
imo this example shuld have both a `discrete` and `continuous` part as youve jus discussed these above hence its thee perfect transition. 

I would not even bother to make parameters for them as its unnecessary repeat from above just make pseudocode a la `SearchSpace(discrete=discrete_space, continuous=continuous_space)`

Reply:
Changed; please have a look if the description is suitable or if it needs rework.

Reply:
dior